src/kafka-util/src/bin/kgen.rs

```diff
@@ -584,16 +615,22 @@ async fn main() -> anyhow::Result<()> {
             eprintln!("Generating message {}", i);
         }
         value_gen.next_value(&mut buf);
-        let key_i = if let Some(key_dist) = key_dist.as_ref() {
-            key_dist.sample(&mut rng)
+        let key = if let Some(key_gen) = key_gen.as_mut() {
+            // TODO: shitty
```

Comment:
Why is it shitty? My guess is because of the allocation of `key_buf` is not reused each time through the loop ?

You could hoist the vector out of the loop and use this kind of pattern (sorry, I tried writing something cleaner involving `Cow` but didn't succeed)

```
fn f(x: bool) {
    let mut v = vec![];
    let mut _bytes_holder = None;
    let to_write: &[u8] = if x {
        v.push(4); v.push(6); v.push(8);
        &v
    } else {
        _bytes_holder = Some(42_i32.to_be_bytes());
        &_bytes_holder.as_ref().unwrap()[..]
    };
}```

Reply:
yeah thats why its shitty. i think using https://docs.rs/byteorder/1.4.2/byteorder/trait.ByteOrder.html#tymethod.write_u64 might be simpler and better?

Reply:
i documented why its annoying id love to merge as-is and then fix the buffer management later if thats ok with you?

Reply:
Eliminating the allocation is a nice-to-have but this is an external tool, it's not like it is perf critical code. Merge away!